sympy/solvers/solveset.py

```diff
@@ -25,95 +25,125 @@
 from sympy.matrices import Matrix
 from sympy.polys import (roots, Poly, degree, together, PolynomialError,
                          RootOf)
-from sympy.solvers.solvers import checksol, denoms
+from sympy.solvers.solvers import checksol, denoms, unrad
 from sympy.solvers.inequalities import solve_univariate_inequality
 from sympy.utilities import filldedent
 
 import warnings
 
 
-def invert_real(f_x, y, x):
-    """ Inverts a real valued function
-
-    Reduces the real valued equation ``f(x) = y`` to a set of equations ``{g(x)
-    = h_1(y), g(x) = h_2(y), ..., g(x) = h_n(y) }`` where ``g(x)`` is a simpler
-    function than ``f(x)``.  The return value is a tuple ``(g(x), set_h)``,
-    where ``g(x)`` is a function of ``x`` and ``set_h`` is the set of
-    functions ``{h_1(y), h_2(y), ..., h_n(y)}``.
+def _invert(f_x, y, x, domain=S.Complexes):
+    """
+    Reduce the complex valued equation ``f(x) = y`` to a set of equations
+    ``{g(x) = h_1(y), g(x) = h_2(y), ..., g(x) = h_n(y) }`` where ``g(x)`` is
+    a simpler function than ``f(x)``.  The return value is a tuple ``(g(x),
+    set_h)``, where ``g(x)`` is a function of ``x`` and ``set_h`` is
+    the set of function ``{h_1(y), h_2(y), ..., h_n(y)}``.
     Here, ``y`` is not necessarily a symbol.
 
-    The ``set_h`` contains the functions along with the information about their
-    domain in which they are valid, through set operations. For instance, if
-    ``y = Abs(x) - n``, is inverted, then, the ``set_h`` doesn't simply
-    return `{-n, n}`, as it doesn't explicitly mentions about the nature of
-    `n` rather it will return:
+    The ``set_h`` contains the functions along with the information
+    about their domain in which they are valid, through set
+    operations. For instance, if ``y = Abs(x) - n``, is inverted
+    in the real domain, then, the ``set_h`` doesn't simply return
+    `{-n, n}`, as the nature of `n` is unknown; rather it will return:
     `Intersection([0, oo) {n}) U Intersection((-oo, 0], {-n})`
 
+    By default, the complex domain is used but note that inverting even
```

Comment:
Thanks for the docstring, it explains the differences between the real and complex domain very clearly.